Foody/Onboarding/Register/RegisterViewController.swift

```diff
@@ -6,24 +6,161 @@
+        
+      
+        let toolbar = UIToolbar()
+        toolbar.sizeToFit()
+        
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: nil, action: #selector(doneTapped))
+        toolbar.setItems([doneButton], animated: true)
+        dateText.inputAccessoryView = toolbar
+        
+        dateText.inputView = datePicker
+        datePicker.datePickerMode = .date
+    }
+    @objc func doneTapped () {
+        
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .none
+        
+        dateText.text = formatter.string(from: datePicker.date)
+        self.view.endEditing(true)
+    }
+    
+    func formatDate(date: Date) -> String {
+        
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd/MM/yyyy"
+        return formatter.string(from: date)
+    }
+    
+    func dateDayCount(date:String,format:String = "dd/MM/yyyy") -> Int {
+        
+        let dateFormatter = DateFormatter()
+        
+        dateFormatter.locale = Locale.current
+        dateFormatter.timeZone = TimeZone.autoupdatingCurrent
+        dateFormatter.timeStyle = .short
+        dateFormatter.dateFormat = format
+        
+        let calender = Calendar.current
+        let firstDate = calender.startOfDay(for: dateFormatter.date(from: date)!)
+        let twiceDate = calender.startOfDay(for: Date())
+        let companents = calender.dateComponents([.year], from: firstDate, to: twiceDate)
+        
+        if companents.year! >= 15 {
+            
+            makeAlert(title: "Registration Successful.", message:"You are welcome. :)" )
+            
+            //Navigation
+            
+        }else {
+            makeAlert(title: "Warning!", message: "Persons under the age of 15 cannot enter.")
+        }
+        return companents.year!
+    }
+    
+    @IBAction func signUpClicked(_ sender: Any) {
+        
+        // email and password validation
+        
+        if let username = usernameText.text , let email = emailText.text , let password = passwordText.text {
+            if username == "" && email == "" && password == "" {
+                
+                makeAlert(title: "Warning!", message: "Please Enter Username, Email and Password!" )
+            }else {
+                if !email.isValidEmail(email: email) {
+                    makeAlert(title: "Warning!", message: "Please Enter Valid Email!")
+                    
+                }else if !password.isValidPassword(password: password){
+                    makeAlert(title: "Warning!", message: "Please Enter Valid Password!")
+                    
+                }else {
+                    
+                    if emailText.text != "" && passwordText.text != "" && dateDayCount(date: dateText.text!) > 15 {
+                        
+                        Auth.auth().createUser(withEmail: emailText.text!, password: passwordText.text!) { (authdata,error) in
```

Comment:
Remove exclamation mark(!) Try to find solution without it. emailText.text! and  passwordText.text!